test/e2e/antreapolicy_test.go

```diff
@@ -1300,16 +1296,23 @@ func testACNPClusterGroupRefRuleIPBlocks(t *testing.T) {
 		}
 		return ip + "/128"
 	}
-	cgName := "cg-ipblock-pod-in-ns-x"
-	cgBuilder := &ClusterGroupSpecBuilder{}
-	var ipBlock []crdv1alpha1.IPBlock
+	var ipBlock1, ipBlock2 []crdv1alpha1.IPBlock
 	for i := 0; i < len(podXAIP); i++ {
-		ipBlock = append(ipBlock, crdv1alpha1.IPBlock{CIDR: genCIDR(podXAIP[i])})
-		ipBlock = append(ipBlock, crdv1alpha1.IPBlock{CIDR: genCIDR(podXBIP[i])})
-		ipBlock = append(ipBlock, crdv1alpha1.IPBlock{CIDR: genCIDR(podXCIP[i])})
-	}
-	cgBuilder = cgBuilder.SetName(cgName).
-		SetIPBlocks(ipBlock)
+		ipBlock1 = append(ipBlock1, crdv1alpha1.IPBlock{CIDR: genCIDR(podXAIP[i])})
+		ipBlock1 = append(ipBlock1, crdv1alpha1.IPBlock{CIDR: genCIDR(podXBIP[i])})
+		ipBlock1 = append(ipBlock1, crdv1alpha1.IPBlock{CIDR: genCIDR(podXCIP[i])})
+		ipBlock2 = append(ipBlock2, crdv1alpha1.IPBlock{CIDR: genCIDR(podZAIP[i])})
+	}
+
+	cgv1a3Name := "cg-v1alpha3-ipblocks-pod-in-ns-x"
+	cgBuilder := &ClusterGroupV1Alpha3SpecBuilder{}
+	cgBuilder = cgBuilder.SetName(cgv1a3Name).
+		SetIPBlocks(ipBlock1)
+	// crd/v1alpha2 ClusterGroups should be converted to crd/v1alpha3.
+	cgv1a2Name := "cg-v1alpha2-ipblock-pod-za"
+	cgBuilder2 := &ClusterGroupV1Alpha2SpecBuilder{}
+	cgBuilder2 = cgBuilder2.SetName(cgv1a2Name).
+		SetIPBlocks(ipBlock2)
```

Comment:
personally I feel like this should be tested separately, not as part of a network policy test: create a v1alpha2 group and retrieve it using the v1alpha3 version, and vice versa

the network policy tests can deal with a single version (the most recent one)